src/v/kafka/server/handlers/incremental_alter_configs.cc

```diff
@@ -110,6 +110,20 @@ static void parse_and_set_shadow_indexing_mode(
     }
 }
 
+static void parse_and_set_topic_replication_factor(
+  cluster::property_update<std::optional<cluster::replication_factor_type>>&
+    property,
+  const std::optional<ss::sstring>& value,
+  config_resource_operation op) {
+    // set property value
+    if (op == config_resource_operation::set) {
+        property.value = boost::lexical_cast<cluster::replication_factor_type>(
```

Comment:
Is there something gracefully catching the exceptional case where the value contains something that doesn't parse as an integer?  clearly it's fine to return an error to the user, but let's make sure it's a well formed error.

Reply:
it is catch blog inside `create_topic_properties_update`